filebeat/input/filestream/metrics_integration_test.go

```diff
@@ -38,9 +38,20 @@ func TestFilestreamMetrics(t *testing.T) {
 		"prospector.scanner.check_interval":    "24h",
 		"close.on_state_change.check_interval": "100ms",
 		"close.on_state_change.inactive":       "2s",
+		"parsers": []map[string]interface{}{
+			{
+				"multiline": map[string]interface{}{
+					"type":      "pattern",
+					"pattern":   "^multiline",
+					"negate":    true,
+					"match":     "after",
+					"max_lines": 1,
```

Comment:
This would not be enough to test it, we truncate for 2 reasons:

* the whole message exceeds [`message_max_bytes`](https://www.elastic.co/guide/en/beats/filebeat/current/filebeat-input-filestream.html#_message_max_bytes)
* when the multiline parser is active and the `max_lines` is exceeded (what you already test for)

We need to test both cases separately and combined.

Reply:
I added individual tests for `multiline.max_lines` and `message_max_bytes` as suggested.

Reply:
Looks like the truncate_fields processor also sets the truncated flag, so it inherits the metric by default. I'll add a test specific for this.

Reply:
@mauri870 if you don't find a way to implement it, we should add comments around metrics mentioning that it does not include any changes coming from processors.

Reply:
Ack, will update the docs mentioning it.